### PR TITLE
feat: integrate delete api into task deletion drawer

### LINKED ISSUE #282: Integrate `DELETE` API into `TaskDeletionDrawer`
## Description

Integrate the existing `DELETE /api/task/:taskId` into the [`TaskDeletionDrawer`](https://github.com/Tascurator/tascurator-frontend/blob/develop/src/components/ui/drawers/TaskDeletionDrawer.tsx) component to enable deleting an existing task. Please ensure that the UI correctly reflects the updates and handles any necessary user interactions with toasts for confirmations and error messages.

## Criteria

- [x] Modify [`TaskDeletionDrawer`](https://github.com/Tascurator/tascurator-frontend/blob/develop/src/components/ui/drawers/TaskDeletionDrawer.tsx) component to deleting an existing task by calling `DELETE /api/task/:taskId`
- [x] Display a success toast when it's completed without errors
- [ ] Display an error toast with an error from the backend when it fails
- [x] If applicable, delete hard-coded code like to simulate data mutation in the previous development

### PR DESCRIPTION
**Pull request by yuki-92c**
## Overview

- I integrated the delete API to `TaskDeletionDrawer` component to enable deleting an existing task.

## Changes

- `src/components/ui/accordion/AccordionTaskItem.tsx`
- `src/components/ui/drawers/TaskDeletionDrawer.tsx`

## Review points

- Delete an task
- Display a toast

## Screenshots or videos

https://github.com/user-attachments/assets/96768c8f-89dd-43ea-adb3-210378ecbf94

## Assignee Checklist:

- [ ] The base branch is correct (no accidental merges)
- [ ] The branch name follows our branch naming rules
- [ ] The PR title follows our PR title rules
- [ ] My code follows our coding style
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [x] Code readability and simplicity
- [x] Follows best practices and coding standards
- [x] Understandable and maintainable code

### CODE REVIEW

File: src/components/ui/drawers/TaskDeletionDrawer.tsx [hunk 1]
```diff
@@ -52,39 +56,48 @@ export const TaskDeletionDrawer = ({
...
-  const [isLoading, setIsLoading] = useState(false);
+  const {
+    handleSubmit,
+    formState: { isSubmitting },
+  } = useForm();
 
-  // TODO: Implement the delete click functionality
   const onSubmit = async () => {
-    setIsLoading(true);
     setOpen(true);
```
Comment (akiomatic) [inline src/components/ui/drawers/TaskDeletionDrawer.tsx]: Could you remove this line as it's not necessary?

File: src/components/ui/drawers/TaskDeletionDrawer.tsx [hunk 2]
```diff
@@ -52,39 +56,48 @@ export const TaskDeletionDrawer = ({
...
+      }
     }
   };
 
   return (
     <>
-      {isLoading ? <LoadingSpinner isLoading={true} /> : ''}
-      <Drawer open={open} onOpenChange={setOpen} modal={!isLoading}>
+      {isSubmitting ? <LoadingSpinner isLoading={true} /> : ''}
+      <Drawer open={open} onOpenChange={setOpen}>
```
Comment (akiomatic) [inline src/components/ui/drawers/TaskDeletionDrawer.tsx]: Could you re-add `modal={!isSubmitting}`?

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you. I've left 2 comments.

File: src/components/ui/drawers/TaskDeletionDrawer.tsx (hunk 2 above)
Comment (yuki-92c) [reply src/components/ui/drawers/TaskDeletionDrawer.tsx]: Ohhh thank you for letting me know

**Review by akiomatic — APPROVED** [review]
LGTM. Thank you for the changes!

**Pull request merged**